### PR TITLE
test: Migrate ListView tests from Enzyme to RTL and improve coverage

### PR DESCRIPTION
**Pull request by vishvamsinh28**
Migrated <ListView /> tests from Enzyme to React Testing Library, added missing edge case validations, improved scroll behavior checks, and ensured internal methods and dynamic rendering logic are fully covered.

### CODE REVIEW

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js [hunk 1]
```diff
@@ -13,231 +13,174 @@
...
+    cleanup();
+    jest.restoreAllMocks();
+  });
 
-    it('renders without exploding', () => {
-      expect(wrapper).toBeDefined();
+  it('renders initial number of items with correct keys and styles', () => {
+    const { container } = render(<ListView {...defaultProps} />);
+    const items = container.querySelectorAll('[data-item-key]');
+    expect(items.length).toBeGreaterThanOrEqual(defaultProps.initialDraw);
```
[the remark this answers is not in the document]
Comment (vishvamsinh28) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js]: Using toBeGreaterThanOrEqual since ListView may render more than initialDraw due to viewBuffer logic after mount.

**Comment by vishvamsinh28** [comment]
@yurishkuro please review

**Review by yurishkuro — COMMENTED** [review]
once again, all method titles are changed, I cannot compare new and previous logic to validate the change

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js [hunk 2]
```diff
@@ -117,7 +144,6 @@ describe('<ListView>', () => {
       let oldInitWrapper;
       const initWrapperMock = jest.fn(elm => {
         if (elm != null) {
-          // jsDom requires `defineProperties` instead of just setting the props
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js]: please keep these comments

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js [hunk 3]
```diff
@@ -132,8 +158,6 @@ describe('<ListView>', () => {
 
       beforeAll(() => {
         oldRender = ListView.prototype.render;
-        // `_initWrapper` is not on the prototype, so it needs to be mocked
-        // on each instance, use `render()` as a hook to do that
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js]: please keep these comments, they seem relevant to explaining why the test is doing this

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js [hunk 4]
```diff
@@ -149,30 +173,49 @@ describe('<ListView>', () => {
...
+          });
+          return <ListView {...props} ref={ref} />;
+        }
+        render(<TestComponent />);
+        instance = componentInstance;
       });
 
       it('getViewHeight() returns the viewHeight', () => {
-        expect(instance.getViewHeight()).toBe(clientHeight);
+        if (instance) {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js]: I don't understand the if(). Shouldn't the test fail otherwise?

**Comment by vishvamsinh28** [comment]
@yurishkuro tests updated

**Review by yurishkuro — APPROVED** [review]

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js [hunk 5]
```diff
@@ -184,59 +226,79 @@ describe('<ListView>', () => {
...
-      it('calls _positionList when the document is scrolled', done => {
-        const event = new Event('scroll');
-        const fn = jest.spyOn(instance, '_positionList');
-        expect(instance._isScrolledOrResized).toBe(false);
-        window.dispatchEvent(event);
-        expect(instance._isScrolledOrResized).toBe(true);
-        window.requestAnimationFrame(() => {
-          expect(fn).toHaveBeenCalled();
+      it('triggers _positionList after scroll when windowScroller is enabled', done => {
+        if (instance) {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js]: this is a bad practice.

Comment (vishvamsinh28) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js]: removed all the if blocks

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js [hunk 6]
```diff
@@ -219,6 +276,7 @@ describe('<ListView>', () => {
       });
 
       it('uses the root HTML element to determine if the view has changed', () => {
+        expect(instance).toBeDefined();
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/ListView/index.test.js:279]: for the future, this seems redundant. If it's not defined the test will still fail on null ptr dereference in the next line.

**Review by yurishkuro — APPROVED** [review]
